src/tests/compiler/evaluate.rs

```diff
@@ -74,9 +73,7 @@ fn test_basic_expand_macro_3() {
 fn convert_clvm_to_chialisp(s: String) -> Result<Rc<SExp>, CompileErr> {
     let loc = Srcloc::start(&"*program*".to_string());
     parse_sexp(loc.clone(), s.bytes())
-        .map_err(|e| {
-            return CompileErr(e.0.clone(), e.1.clone());
-        })
+        .err_into()
```

Comment:
I don't think you need this. I think you could either put a `?` here, to have the error be returned immediately. the `?`-operator will apply conversions as long as one of `From` or `Into` are implemented for the error types. But I think you could also just let the error propagate through this expression and exit at the end, by putting the `?` after the last call to `map()`.
I would expect `map()` to only apply to the `Ok` value, not to the `Err`.

Reply:
It is needed because the type system can't figure out which of multiple alternative error types to use for the whole expression before applying .from at the point of ?.  I tested this.  You do need an explicit conversion.

Reply:
I see what you're doing now. You need the error code (and the correct `Result` type) to flow through the `.and_then()`. Rust doesn't provide quite the same language facilities to have that happen, compared to (in my mind, more straight forward) series of plain statements. If you would you would get implicit conversion of the error type if you would write this block of code like this:

```
    let parsed_program = parse_sexp(loc.clone(), s.bytes())?;
    let program = frontend(opts.clone(), &parsed_program)?;
    let e = Evaluator::new(opts.clone(), runner, program.helpers);
    let result = e.shrink_bodyform(
        &mut allocator,
        program.args.clone(),
        &HashMap::new(),
        program.exp.clone(),
        false,
        Some(EVAL_STACK_LIMIT),
    )?;
    Ok(result.to_sexp().to_string())
```